packages/meditrak-app-server/src/routes/AuthRoute.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+
+import { Route } from '@tupaia/server-boilerplate';
+import { Resolved } from '@tupaia/tsutils';
+import { TupaiaApiClient } from '@tupaia/api-client';
+
+type RefreshTokenBody = {
+  refreshToken: string;
+};
+
+type ReAuthenticateBody = {
+  emailAddress: string;
+  password: string;
+  deviceName: string;
+  devicePlatform: string;
+  installId: string;
+};
+
+export type AuthRequest = Request<
+  {
+    grantType?: 'refresh_token';
+  },
+  | Resolved<ReturnType<TupaiaApiClient['auth']['login']>>
+  | Resolved<ReturnType<TupaiaApiClient['auth']['refreshAccessToken']>>,
```

Comment:
That caught me at the beginning, I thought it is another syntax of TypeScript. I feel it would be less confusing if we add a `getAuth()`.

Reply:
Which part's confusing? The `|` symbol at the start of both lines? Yeah I find it pretty weird too... vscode just does it.

I can create a separate union type for that 👍

Reply:
Ah it didn't highlight it. This part `TupaiaApiClient['auth']['refreshAccessToken']`

Reply:
Okay scratch that, even doing it as a separate type has the `|` at the start. I think we may just have to get used to it?